src/Popover/Popover.tsx

```diff
@@ -0,0 +1,145 @@
+import React, {
+  Ref,
+  FunctionComponent,
+  ReactNode,
+} from 'react';
+import styled from 'styled-components';
+import { fromTheme } from 'mark-one';
+
+/**
+ * Expected props for a Popover Component
+ */
+
+interface PopoverProps {
+  /**
+   * The content that will appear inside the body of the popover
+   */
+  children: ReactNode;
+  /**
+   * An optional passthrough className to allow restyling the Popover
+   * Will be applied to the PopoverWrapper, allowing for cascading styles to be
+   * applied to the internal components
+   */
+  className?: string;
+  /**
+   * An optional forwarded Ref that will be attached to the PopoverBody, e.g.
+   * for focusing the content
+   */
+  forwardRef?: Ref<HTMLDivElement>;
+  /**
+   * Control whether the Popover should be displayed on the page.
+   */
+  isVisible: boolean;
+  /**
+   * The optional content that will appear as the heading of the Popover
+   */
+  title?: ReactNode;
+  /**
+   * The Optional  offset that should be applied to the left side of the the
+   * Popover element, relative to its placement in the DOM.
+   * Should specified in pi, em, rem, % or 0.
+   */
+  xOffset?: string;
+  /**
+   * The Optional  offset that should be applied to the bottom of the the
+   * Popover element, relative to its placement in the DOM.
+   * Should specified in pi, em, rem, % or 0.
+   */
+  yOffset?: string;
+}
+
+/**
+ * Sets the amount of offset that should be applied to the Popover element
+ */
+interface PopoverWrapperProps {
+  /**
+   * Sets the position of the left edge of the Popover
+   */
+  left: PopoverProps['xOffset'];
+  /**
+   * Sets the position of the bottom edge of the Popover
+   */
+  bottom: PopoverProps['yOffset'];
+}
+
+const PopoverPlaceholder = styled.div`
```

Comment:
What's the purpose of this element? Is it solely to provide an "anchor point" (for want of a better term) to apply the `className` prop to?

Reply:
Yeah, it's basically an anchor that makes the {x|y}Offset props move the Popover from its place in the DOM. Basically the `position: absolute` setting on the `PopoverWrapper` requires a positioned parent element, otherwise it applies the positional properties (`left`, `bottom`) from the body of the page instead.

Reply:
Gotcha. Thanks for the explanation!